book/integration-service.md

```diff
@@ -0,0 +1,264 @@
+- Finally, if any automatic ReleasePlans have been created by the user in the workspace, it will create a Release CR signalling intent to release the tested content, to be carried out by the [release-service](./release-service.md).
+
+The diagram below shows the interaction of the integration service and other services.
+
+![](../diagrams/hacbs-data-flow.jpg)
+
+### Dependencies
+
+The [Integration Service](./integration-service.md) is dependent on the following services:
+- [Pipeline Service](./pipeline-service.md)
+  - Pipeline execution, Pipeline logging
+- [GitOps Service](./gitops-service.md)
+  - Provides the facility to create
+    - Snapshots defining sets of Builds to test
+    - Environment to test the Application on
+    - SnapshotEnvironmentBindings to have the Snapshot of the Application deployed to a specific Environment
+- [Hybrid Application Service](./hybrid-application-service.md)
+  - Provides the Application and Component model. Integration Service updates the pullspec reference on the Component CR when a component passes its required testing.
+- [Release Service](./release-service.md)
+  - Provides the ReleasePlan that will be used to determine if integration-service should create a Release
+- [Enterprise Contract Service](./enterprise-contract.md)
+  - Provides facilities to validate whether content has passed the Enterprise Contract.
+
+
+## Terminology
+
+**Component Pipeline** - This is the test pipeline run for the Component of an Application that gets triggered by a completed Build pipeline.
+
+**Composite Pipeline** - This is the pipeline run that runs if two or more Components have completed and passed their Component pipeline concurrently so that they should be tested together.
+
+**Global Candidate List** - The list of all Component digest that have passed individual component testing. This can be retrieved from the Application to see what Components it is made of and then querying each of the Components `spec.containerImage`. 
+
+**Snapshot** - The list that maps all Components of an Application to their Component Image digests. Once created, the snapshot is immutable. 
+
+**IntegrationTestScenario** - The custom resource that describes separate PipelineRuns that are supposed to be run by the snapshot controller. Any number of IntegrationTestScenarios can be set by the user and each of them includes information about the Tekton pipeline that will be executed - Tekton bundle, pipeline name and parameters, and the Environment details. The Integration Test Scenarios can also specify the contexts where they will be applied - this can be used to define when the scenario will run (Component or Composite stage - or both). If more than one Integration Test Scenario has the same context, their respective PipelineRuns will be executed in parallel.
```

Comment:
We can also add this sentence at the end:
```
The IntegrationTestScenarios can be marked as optional, in which case they will not be taken into account when determining if the Snapshot has passed testing.
```

Reply:
Done

Reply:
Thanks, looks good